### ISSUE TITLE
do not transpile, only check types

### ISSUE THREAD

**Issue by artin-phares** [s-panferov/awesome-typescript-loader#520]
It is possible since typescript v2.3 to `--checkJs` to fetch types info from `jsdoc` annotation.  
So with `typescript` CLI I can do
```
tsc --allowJs --checkJs --noEmit normal.js
```
This just checks `normal.js` without transpilation. Later I can transpile it with `babel`.  

In the result - I'm not constrained to use only features implemented by `typescript`, but can use any new ecmascript proposals (like [class private fields](https://github.com/tc39/proposal-private-methods)). Same time I have static type checks.

All looks good so far, but I cannot get this work with webpack loader.  
Eg. with class private fields, loader first transpiles class into function, so babel's `transform-private` does not see any private class fields to transform.  

I need `awesome-typescript-loader` to be 'transparent' (do not modify, do not produce anything), same as CLI  `--noEmit`, to later pipe unmodified code to `babel-loader`

Do you think it is reasonable for loader to have option like `checkOnly` to skip transpilation completely?
If not, any known workarounds?

**Comment by s-panferov** [comment]
@artin-phares take a look at https://github.com/Realytics/fork-ts-checker-webpack-plugin

**Comment by artin-phares** [comment]
@s-panferov

Yep, looks like `fork-ts-checker-webpack-plugin` solves my case. Closing.

---

Btw, in case I still wanna use few `.ts` files (typescript not fully supports `jsdoc` yet) and transpile them with the awesome loader. I've noticed that loader does not respect `// @ts-nocheck`, in contrast with the plugin.  

Eg. I have `.js` file with `@decorator` syntax and `// @ts-nocheck` comment in header. Loader targets `.ts` files only, `experimentalDecorators` not switched on (intentionally).  
But I'm still receiving `ERROR in [at-loader]` from that file (with `// @ts-nocheck`), while there are no errors from `fork-ts-checker-webpack-plugin`.  
 
Any ideas why this happens?
